### ISSUE TITLE
[console] Module with ServiceProvider class issues

### ISSUE THREAD

**Issue by dpagini** [hechoendrupal/drupal-console#3769]
### Problem/Motivation
Would anyone be able to help me with an issue... I think this is above my head. I am trying to use drupal console, but I have a module called `config_override` that's throwing a fatal error, only in drupal console. So I know it's probably a fix that THAT module needs, but just wondering if any knowledge on this project would be able to help me figure out what's going on...?

The `config_override` module uses a ServiceProvider to dynamically set a container parameter. Then it defines a new service in that ServiceProvider, and passes the parameter as an argument.

In drupal console, the error is that the service constructor is expecting an array, but it's getting the `new Parameter('config_override__environment')` object passed to it instead...

```sh
vagrant@local:/var/www/mysite/docroot$ drupal
TypeError: Argument 1 passed to Drupal\config_override\EnvironmentConfigOverride::__construct() must be of the type array, object given in /var/www/mysite/docroot/modules/contrib/config_override/src/EnvironmentConfigOverride.php on line 39 #0 [internal function]: Drupal\config_override\EnvironmentConfigOverride->__construct(Object(Symfony\Component\DependencyInjection\Parameter))
#1 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(905): ReflectionClass->newInstanceArgs(Array)
#2 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(451): Symfony\Component\DependencyInjection\ContainerBuilder->createService(Object(Symfony\Component\DependencyInjection\Definition), 'config_override...')
#3 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(962): Symfony\Component\DependencyInjection\ContainerBuilder->get('config_override...', 1)
#4 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(959): Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices(Object(Symfony\Component\DependencyInjection\Reference))
#5 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(1150): Symfony\Component\DependencyInjection\ContainerBuilder->resolveServices(Array)
#6 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(923): Symfony\Component\DependencyInjection\ContainerBuilder->callMethod(Object(Drupal\Core\Config\ConfigFactory), Array)
#7 /var/www/mysite/vendor/symfony/dependency-injection/ContainerBuilder.php(451): Symfony\Component\DependencyInjection\ContainerBuilder->createService(Object(Symfony\Component\DependencyInjection\Definition), 'config.factory')
#8 /var/www/mysite/docroot/core/lib/Drupal.php(369): Symfony\Component\DependencyInjection\ContainerBuilder->get('config.factory')
#9 /var/www/mysite/docroot/core/modules/system/system.module(1110): Drupal::config('core.extension')
#10 /var/www/mysite/vendor/drupal/console/src/Extension/Manager.php(226): system_rebuild_module_data()
#11 /var/www/mysite/vendor/drupal/console/src/Extension/Manager.php(148): Drupal\Console\Extension\Manager->discoverExtensions('module')
#12 /var/www/mysite/vendor/drupal/console/src/Extension/Manager.php(115): Drupal\Console\Extension\Manager->discoverExtension('module')
#13 /var/www/mysite/vendor/drupal/console/src/Utils/AnnotationValidator.php(75): Drupal\Console\Extension\Manager->discoverModules()
#14 /var/www/mysite/vendor/drupal/console/src/Utils/AnnotationValidator.php(60): Drupal\Console\Utils\AnnotationValidator->isExtensionInstalled('rest')
#15 /var/www/mysite/vendor/drupal/console/src/Application.php(121): Drupal\Console\Utils\AnnotationValidator->isValidCommand('Drupal\\Console\\...')
#16 /var/www/mysite/vendor/drupal/console/src/Application.php(62): Drupal\Console\Application->validateCommands()
#17 /var/www/mysite/vendor/symfony/console/Application.php(125): Drupal\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /var/www/mysite/vendor/drupal/console/bin/drupal.php(88): Symfony\Component\Console\Application->run()
#19 phar:///usr/local/bin/drupal/src/Utils/Launcher.php(30): include_once('/var/www/mysite...')
#20 phar:///usr/local/bin/drupal/bin/drupal.php(71): Drupal\Console\Launcher\Utils\Launcher->launch(Object(Drupal\Console\Core\Utils\DrupalFinder))
#21 phar:///usr/local/bin/drupal/bin/drupal(3): require('phar:///usr/loc...')
#22 /usr/local/bin/drupal(10): require('phar:///usr/loc...')
#23 {main}
TypeError: Argument 1 passed to Drupal\config_override\EnvironmentConfigOverride::__construct() must be of the type array, object given in Drupal\config_override\EnvironmentConfigOverride->__construct() (line 39 of /var/www/mysite/docroot/modules/contrib/config_override/src/EnvironmentConfigOverride.php).
```

**Details to include:**
- Drupal version 8.4.4
- Console version 1.5.0

### Solution
?? Looking for help to understand... is this a console error, or a `config_override` error? My site works just fine, it's just console that reports this as a problem.

**Comment by dpagini** [comment]
Hmm... posted a patch over at `config_override` module. Not sure if this is the right fix still though, so if anyone can comment on why/how this is happening, that would be very helpful!

https://www.drupal.org/project/config_override/issues/2943061

**Comment by dpagini** [comment]
Ping @jmolivas - ever have any thoughts on this one?

**Comment by enzolutions** [comment]
@dpagini the error stills exist in Drupal 8.7.x?

**Comment by gilbertomangones** [comment]
Hi, I tried the module ```config_override``` on Drupal ```Drupal core 8.7.7```

1. Applied ```drupalspanish mac$ composer require drupal/config_override```
![2019-09-18_1813](https://user-images.githubusercontent.com/6983125/65193270-20872280-da40-11e9-97a2-8c0e91b3d311.png)

2.
 ```$ cd web/modules/contrib/``` 
```$ ls -ls```
total 0
0 drwxr-xr-x  9 mac  admin  288 Sep 18 17:48 config_override

**Comment by enzolutions** [comment]
Looks like the issue is solved as tested in version 1.9.3
